PySpice/Spice/HSpice/Server.py

```
@@ -0,0 +1,349 @@
+        path = os.path.join(lock_dir, f"lock_{i}")
+        fd = os.open(path, os.O_RDWR | os.O_CREAT, 0o600)
+        lock_files.append((fd, i))
+
+    while True:
+        for fd, idx in lock_files:
+            try:
+                fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                for other_fd, other_idx in lock_files:
+                    if other_idx != idx:
+                        os.close(other_fd)
+                return fd, idx
```

Ruff (0.15.20)

[warning] Consider moving this statement to an `else` block

(TRY300)